File: src/main/java/k31/grc/ast/node/type/DimIntegerT.java
package k31.grc.ast.node.type;

import k31.grc.ast.node.NodeBase;
import k31.grc.ast.visitor.Visitor;

public class DimIntegerT extends NodeBase {

	private Integer dim;

	public Integer getDim() {
		return dim;
	}

	public DimIntegerT(String text) {
		super(text);

		this.dim = Integer.parseInt(text.replace("[", "").replace("]", ""));
	}

	@Override
	public void accept(Visitor v) {

		v.visit(this);
	}
}

File: DotNetGrc/Grc/Ast/Node/Type/DimIntegerT.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Grc.Ast.Visitor;

namespace Grc.Ast.Node.Type
{
	public class DimIntegerT : NodeBase
	{
		private readonly string integer;

		private readonly string lbrack;
		private readonly string rbrack;

		private readonly int line;
		private readonly int pos;

		public string Integer { get { return integer; } }

		public override int Line { get { return line; } }

		public override int Pos { get { return pos; } }

		public DimIntegerT(string integer, string lbrack, string rbrack, int line, int pos)
		{
			this.integer = integer;
			this.lbrack = lbrack;
			this.rbrack = rbrack;

			this.line = line;
			this.pos = pos;
		}

		public override void Accept(IVisitor v)
		{
			v.Visit(this);
		}

		public override string Text
		{
			get { return string.Format("{0}{1}{2}", lbrack, integer, rbrack); }
		}
	}
}
